hibernate-core/src/test/java/org/hibernate/test/hql/size/SizeTest.java

```diff
@@ -102,6 +113,31 @@ public void testSizeWithoutNestedPath() {
 		} );
 	}
 
+	@Test
+	public void testSizeWithoutNestedPathAddFetch() {
+		inTransaction( session -> {
+			List<Teacher> teachers = session.createQuery(
+					"select teacher from Teacher teacher left join fetch teacher.skills where size(teacher.skills) > 0",
+					Teacher.class
+			).getResultList();
+			assertEquals( countNumberOfJoins( sqlStatementInterceptor.getSqlQueries().get( 0 ) ), 2 );
+			assertEquals( 2, teachers.size() );
+		} );
+	}
+
+	@Test
+	public void testSizeWithoutNestedPathAddFetchDistinct() {
+		inTransaction( session -> {
+			List<Teacher> teachers = session.createQuery(
+					"select distinct teacher from Teacher teacher left join fetch teacher.skills where size(teacher.skills) > 0",
+					Teacher.class
+			).getResultList();
+			assertEquals( countNumberOfJoins( sqlStatementInterceptor.getSqlQueries().get( 0 ) ), 2 );
+			assertEquals( 2, teachers.size() );
```

Comment:
The above assertion passes because `distinct` is used in the query. 

I don't believe that using `distinct` should be necessary though. See https://github.com/hibernate/hibernate-orm/pull/3684/commits/9c1f805e53d15c377cb0ded21ba1f859658bc412#r547585526 .